src/gluonts/trainer/_base.py

```diff
@@ -315,26 +323,31 @@ def loop(
                         )  # TODO: handle possible exception
 
                     if not trainer.learning_rate == curr_lr:
-                        logging.info(
+                        if best_epoch_info.epoch_no == -1:
+                            raise GluonTSUserError(
+                                "Got NaN in first epoch. Try reducing initial learning rate."
+                            )
```

Comment:
I'm not a fan of this, but maybe we can improve at a later stage

Reply:
I tried to save the model before the training start and then use that serialized model in this case. The challenge is that we cannot serialize the model before the first batch ran through. So this would complicate things a bit (although maybe not that much).